go/libraries/doltcore/sqle/dtables/column_diff_table.go

```diff
@@ -0,0 +1,649 @@
+		}
+		hashes, commits, metas := index.HashesToCommits(ctx, dt.ddb, hs, dt.head, false)
+		if len(hashes) == 0 {
+			return sql.PartitionsToPartitionIter(), nil
+		}
+
+		headHash, err := dt.head.HashOf()
+		if err != nil {
+			return nil, err
+		}
+		var partitions []sql.Partition
+		for i, h := range hashes {
+			if h == headHash && commits[i] == nil {
+				partitions = append(partitions, newDoltDiffPartition(workingSetPartitionKey))
+			} else {
+				partitions = append(partitions, doltdb.NewCommitPart(h, commits[i], metas[i]))
+			}
+		}
+		return sql.PartitionsToPartitionIter(partitions...), nil
+	}
+
+	return dt.Partitions(ctx)
+}
+
+type doltColDiffWorkingSetRowItr struct {
+	stagedIndex         int
+	unstagedIndex       int
+	colIndex            int
+	stagedTableDeltas   []diff.TableDelta
+	unstagedTableDeltas []diff.TableDelta
+}
+
+func (dt *ColumnDiffTable) newWorkingSetRowItr(ctx *sql.Context) (sql.RowIter, error) {
+	sess := dsess.DSessFromSess(ctx.Session)
+	roots, ok := sess.GetRoots(ctx, dt.dbName)
+	if !ok {
+		return nil, fmt.Errorf("unable to lookup roots for database %s", dt.dbName)
+	}
+
+	staged, unstaged, err := diff.GetStagedUnstagedTableDeltas(ctx, roots)
+	if err != nil {
+		return nil, err
+	}
+
+	var ri sql.RowIter
+	ri = &doltColDiffWorkingSetRowItr{
+		stagedTableDeltas:   staged,
+		unstagedTableDeltas: unstaged,
+	}
+
+	for _, filter := range dt.partitionFilters {
+		ri = plan.NewFilterIter(filter, ri)
+	}
+
+	return ri, nil
+}
+
+// incrementColIndex increments the column index and table changes index.  When the end of the column names array is
+// reached, moves to the next table changes delta.
+func (d *doltColDiffWorkingSetRowItr) incrementColIndex(tableDelta diff.TableDelta, staged bool) {
+	d.colIndex++
+	var numCols int
+	if tableDelta.ToSch != nil {
+		numCols = tableDelta.ToSch.GetAllCols().Size()
+	} else {
+		numCols = tableDelta.FromSch.GetAllCols().Size()
+	}
+
+	// move to next table once all modified columns are iterated through
+	if d.colIndex >= numCols {
+		d.colIndex = 0
+		if staged {
+			d.stagedIndex++
+		} else {
+			d.unstagedIndex++
+		}
+	}
+}
+
+func (d *doltColDiffWorkingSetRowItr) Next(ctx *sql.Context) (sql.Row, error) {
+	var changeSet string
+	var tableDelta diff.TableDelta
+	// staged keeps track of whether we are looking at staged changes or working set changes
+	staged := false
+	if d.stagedIndex < len(d.stagedTableDeltas) {
+		changeSet = "STAGED"
+		tableDelta = d.stagedTableDeltas[d.stagedIndex]
+		staged = true
+	} else if d.unstagedIndex < len(d.unstagedTableDeltas) {
+		changeSet = "WORKING"
+		tableDelta = d.unstagedTableDeltas[d.unstagedIndex]
+	} else {
+		return nil, io.EOF
+	}
+
+	defer d.incrementColIndex(tableDelta, staged)
+
+	change, err := tableDelta.GetSummary(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var colName string
+	if tableDelta.FromSch != nil {
+		colName = tableDelta.FromSch.GetAllCols().GetColumnNames()[d.colIndex]
```

Comment:
I think getting modified column names here is not correct. I'm getting some of the unmodified column names as well, in this case, the column was dropped. Here is a test case on it:
```
dolt sql -q "create table test (id int primary key, c1 int, c2 int, c3 int);"
dolt commit -Am "add table"
dolt sql -q "alter table test drop column c2;"
dolt sql -q "select commit_hash, table_name, column_name, data_change, schema_change from dolt_column_diff;"
+----------------------------------+------------+-------------+-------------+---------------+
| commit_hash                      | table_name | column_name | data_change | schema_change |
+----------------------------------+------------+-------------+-------------+---------------+
| WORKING                          | test       | id          | false       | true          |
| WORKING                          | test       | c1          | false       | true          |
| WORKING                          | test       | c2          | false       | true          |
| 2geog1nqcirna32l93nvnpf4vm078bbe | test       | id          | false       | true          |
| 2geog1nqcirna32l93nvnpf4vm078bbe | test       | c1          | false       | true          |
| 2geog1nqcirna32l93nvnpf4vm078bbe | test       | c2          | false       | true          |
| 2geog1nqcirna32l93nvnpf4vm078bbe | test       | c3          | false       | true          |
+----------------------------------+------------+-------------+-------------+---------------+
```

This gave me an idea that it might be helpful to have information about schema changes on columns such as `add`, `drop` and `change/modify` column. A way to add this information can be making `schema_change` column to have enum(`added`,`dropped`,`modified`) values instead of bool value option. Let me know what you think 🙂